portal-impl/src/com/liferay/portal/model/impl/UserImpl.java

```diff
@@ -784,7 +784,7 @@ public boolean isEmailAddressComplete() {
 	@Override
 	public boolean isEmailAddressVerificationComplete() {
 		if (isGuestUser() || isEmailAddressVerified() ||
-			isServiceAccountUser()) {
+			(getType() == UserConstants.TYPE_DEFAULT_SERVICE_ACCOUNT)) {
```

Comment:
Why do we want to check specifically for the default service account? Should this not be the same logic for any service account if it is selectable for client credentials impersonation user?

Reply:
Truth be told, I'm not very familiar with service accounts.  I saw the title of the ticket "default service account does not work correctly with oAuthApplicationHeadlessServer" and assumed the default service account was the only type which should be white-listed with emailVerification and passwordReset.

I'm not entirely sure how to create a service account, which leads me to the conclusion they aren't a standard "user", and probably should bypass these checks, just like the default service account.  I will redo my commits.

Reply:
It might be that the default service account is different in concept to the regular ones. I would check with the team that introduced them to gain a broader understanding. Then afterwards assess if that means a poor / unexpected behaviour for the OAuth 2 Applications configuration if we add a different behaviour for the "default" one only. For example we may need to not allow selection of those regular service accounts.

My expectation of service accounts is that they exist for remote systems to authenticate with strong passwords that can have predictable password rotation and controlled permissions. There might be additional requirements for them also.

Reply:
It looks like service accounts shouldn't need email verification, since they aren't users per-say.  See https://cloud.google.com/iam/docs/service-account-overview

> A service account is a special kind of account typically used by an application or compute workload, such as a Compute Engine instance, **rather than a person**. A service account is identified by its email address, which is unique to the account.